src/storage-preview/azext_storage_preview/tests/latest/test_storage_queue_scenarios.py

```diff
@@ -0,0 +1,249 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.testsdk import (ScenarioTest, ResourceGroupPreparer, StorageAccountPreparer)
+from azure.cli.testsdk.checkers import (JMESPathCheck, JMESPathCheckExists, JMESPathCheckNotExists, NoneCheck)
+from .storage_test_util import StorageScenarioMixin
+
+
+class StorageQueueScenarioTests(StorageScenarioMixin, ScenarioTest):
+    @ResourceGroupPreparer(name_prefix='clitest')
+    @StorageAccountPreparer(name_prefix='queue', kind='StorageV2', location='eastus2', sku='Standard_RAGRS')
+    def test_storage_queue_general_scenario(self, resource_group, storage_account):
+        from datetime import datetime, timedelta
+
+        account_key = self.get_account_key(resource_group, storage_account)
+        connection_string = self.get_connection_string(resource_group, storage_account)
+
+        self.set_env('AZURE_STORAGE_ACCOUNT', storage_account)
+        self.set_env('AZURE_STORAGE_KEY', account_key)
+
+        queue = self.create_random_name('queue', 24)
+
+        # Test create with metadata
+        self.cmd('storage queue create -n {} --fail-on-exist --metadata a=b c=d'.format(queue),
+                 checks=JMESPathCheck('created', True))
+        # Test create with fail-on-exist
+        from azure.core.exceptions import ResourceExistsError
+        with self.assertRaisesRegexp(ResourceExistsError, 'The specified queue already exists.'):
+            self.cmd(
+                'storage queue create -n {} --fail-on-exist --connection-string {}'.format(queue, connection_string))
+
+        # Test exists
+        self.cmd('storage queue exists -n {}'.format(queue),
+                 checks=JMESPathCheck('exists', True))
+
+        # Test generate-sas with start, expiry and permissions
+        start = (datetime.utcnow() + timedelta(hours=1)).strftime('%Y-%m-%dT%H:%MZ')
+        expiry = (datetime.utcnow() + timedelta(hours=2)).strftime('%Y-%m-%dT%H:%MZ')
+        sas = self.cmd('storage queue generate-sas -n {} --permissions r --start {} --expiry {}'
+                       .format(queue, start, expiry)).output
```

Comment:
could you add tests to make sure the generated sas work as expected?
And please also add test with `--policy-name`

Reply:
I have test the generated sas in `test_storage_sas_scenarios.py` which is `LiveScenarioTest`.